pico/configdriven/processor/src/main/java/io/helidon/pico/configdriven/processor/ConfiguredByAnnotationProcessor.java

```diff
@@ -0,0 +1,117 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.configdriven.processor;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import io.helidon.common.types.AnnotationAndValue;
+import io.helidon.common.types.TypeInfo;
+import io.helidon.common.types.TypeName;
+import io.helidon.common.types.TypedElementName;
+import io.helidon.pico.configdriven.api.ConfiguredBy;
+import io.helidon.pico.configdriven.runtime.AbstractConfiguredServiceProvider;
+import io.helidon.pico.processor.PicoAnnotationProcessor;
+import io.helidon.pico.tools.ActivatorCreatorProvider;
+import io.helidon.pico.tools.ServicesToProcess;
+
+import static io.helidon.common.types.AnnotationAndValueDefault.findFirst;
+import static io.helidon.common.types.TypeNameDefault.create;
+import static io.helidon.common.types.TypeNameDefault.createFromGenericDeclaration;
+import static io.helidon.common.types.TypeNameDefault.createFromTypeName;
+import static io.helidon.common.types.TypeNameDefault.toBuilder;
+import static io.helidon.pico.configdriven.processor.ConfiguredByProcessorUtils.createExtraActivatorClassComments;
+import static io.helidon.pico.configdriven.processor.ConfiguredByProcessorUtils.createExtraCodeGen;
+
+/**
+ * Extension to {@link PicoAnnotationProcessor} that will handled {@link io.helidon.pico.configdriven.api.ConfiguredBy} services.
+ */
+public class ConfiguredByAnnotationProcessor extends PicoAnnotationProcessor {
+
+    /**
+     * Service loader based constructor.
+     *
+     * @deprecated this is a Java ServiceLoader implementation and the constructor should not be used directly
+     */
+    @Deprecated
+    public ConfiguredByAnnotationProcessor() {
+        super(true);
+    }
+
+    @Override
+    protected Set<String> supportedServiceClassTargetAnnotations() {
+        Set<String> supported = new LinkedHashSet<>(super.supportedServiceClassTargetAnnotations());
+        supported.add(ConfiguredBy.class.getName());
+        return supported;
+    }
+
+    @Override
+    protected void processExtensions(ServicesToProcess services,
+                                     TypeInfo service,
+                                     Set<TypeName> serviceTypeNamesToCodeGenerate,
+                                     Collection<TypedElementName> allElementsOfInterest) {
+        Optional<? extends AnnotationAndValue> configuredByAnno = findFirst(ConfiguredBy.class, service.annotations());
```

Comment:
There is no reason for the processor to depend on configdriven. I would prefer to use String based lookup (so the classpath of processor is as small as possible)